core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -416,6 +419,39 @@ public void setChannel(InputStream in, OutputStream out, OutputStream launchLog,
         setChannel(channel,launchLog,listener);
     }
 
+    /**
+     * Creates a {@link Channel} from the given Channel Builder and Command Transport.
+     * This method can be used to allow {@link ComputerLauncher}s to create channels not based on I/O streams.
+     *
+     * @param cb
+     *      Channel Builder.
+     *      To print launch logs this channel builder should have a Header Stream defined
+     *      (see {@link ChannelBuilder#getHeaderStream()}) in this argument or by one of {@link ChannelConfigurator}s.
+     * @param commandTransport
+     *      Command Transport
+     * @param listener
+     *      Gets a notification when the channel closes, to perform clean up. Can be null.
+     *      By the time this method is called, the cause of the termination is reported to the user,
+     *      so the implementation of the listener doesn't need to do that again.
+     * @since TODO
+     */
+    public void setChannel(@Nonnull ChannelBuilder cb,
+                           @Nonnull CommandTransport commandTransport,
+                           @Nonnull Channel.Listener listener) throws IOException, InterruptedException {
```

Comment:
@oleg-nenashev `Channel.Listener listener` can be null

Reply:
Yes, the original method says:

```
      *      Gets a notification when the channel closes, to perform clean up. Can be null.
      *      By the time this method is called, the cause of the termination is reported to the user,
     *      so the implementation of the listener doesn't need to do that again.
```

Will change it

Reply:
thanks @oleg-nenashev, I got a compilation error with the build https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fremoting-kafka-plugin/detail/PR-2/15/pipeline/ in https://github.com/jenkinsci/remoting-kafka-plugin/pull/2 because Maven recognized wrong method signature of `setChannel()` when I invoked the method call https://github.com/pvtuan10/remoting-kafka-plugin/blob/feature/master-agent-communication/plugin/src/main/java/io/jenkins/plugins/remotingkafka/KafkaComputerLauncher.java#L46. I think changing this method will resolve the problem.